src/index.ts

```diff
@@ -142,9 +146,23 @@ export function withSpid(
     samlConfig
   );
 
-  return loadSpidStrategyOptions()
+  return fromNullable(null)(appConfig.startupIdpsMetadata)
+    .map(parseStartupSpidStrategy)
+    .fold(
+      () => loadSpidStrategyOptions(),
+      idpOptionsRecord =>
+        new Task(() =>
```

Comment:
you can use `async ()` as a shortcut instead of `Promise.resolve`

Reply:
Fixed in a4bdb13